### PR TITLE
fix: LW-8391 de-registration wrong calculation

### PR DESCRIPTION
**Pull request by xdzurman**
# Checklist

- [x] [JIRA ](https://input-output.atlassian.net/browse/LW-8391?atlOrigin=eyJpIjoiZTMxODcyOTdlZGJmNDdhZmFjMGM1ODhkODc5MTJiOGEiLCJwIjoiaiJ9)
- [ ] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

Use updated sdk's packages, in which `implicitCoin` has both `input` (sum of withdrawals + deposits) and `withdrawals` props. In all places where reclaim deposit is extracted, use this functionality.

## Testing

Wallet: 'butter swear more whip vendor attend hood hire plate tell true reduce cheese athlete spring predict certain fruit mass book dog immune wrist rookie'
Check 14.11.2023's de-registration tx
image
It has withdrawals, but correctly should show only deposit reclaim  
image

Same for tx in the JIRA issue screenshot (08 September 2023)
Before:
image
After:
image

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/2034/6187466466/index.html) for [24392cfa](https://github.com/input-output-hk/lace/pull/530/commits/24392cfa1d274fb69ecfc428c5e246077198cd4d)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 29     | 3      | 0       | 0     | 32    | ❌     |

### CODE REVIEW

File: packages/cardano/src/wallet/util/calculate-deposit-reclaim.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Cardano } from '@cardano-sdk/core';
+
+// SDK's ImplicitCoin's "input" field contains summed up withdrawals and stake key reclaim deposits.
+// This function calculates just the stake key reclaim deposit.
+export const calculateDepositReclaim = ({
+  input,
+  withdrawals
+}: Cardano.util.ImplicitCoin): Cardano.Lovelace | undefined => {
+  if (!input) return undefined;
```
Comment (refi93) [inline packages/cardano/src/wallet/util/calculate-deposit-reclaim.ts:9]: bit hypothetical, but can't the implicit input contain also returned collateral if a smart-contract tx fails 🤔 ? not worth dealing with it right now, but if that's the case perhaps I'd at least acknowledge this edge-case with a comment

**Review by refi93 — APPROVED** [review]

File: packages/cardano/src/wallet/util/calculate-deposit-reclaim.ts (hunk 1 above)
Comment (przemyslaw-wlodek) [reply packages/cardano/src/wallet/util/calculate-deposit-reclaim.ts:9]: @input-output-hk/lace-sdk Can this be a case?

**Review by przemyslaw-wlodek — APPROVED** [review]
Great work @xdzurman! 🍺 😄

File: packages/cardano/src/wallet/util/calculate-deposit-reclaim.ts (hunk 1 above)
Comment (refi93) [reply packages/cardano/src/wallet/util/calculate-deposit-reclaim.ts:9]: @mkazlauskas you are right - indeed, I just checked and collateral return has it's own explicit output in the tx body so there's no issue with it, so all good 👍

https://github.com/input-output-hk/cardano-ledger/blob/83cdef1643efe86a55be91bab25b1c81db11a280/eras/babbage/test-suite/cddl-files/babbage.cddl#L68

**Pull request merged**